[MLIR][TOSA] Add tosa.slice operation conversion failure scenario

LiqinWeng opened the pull request:
Fixes #68481, In the following scenario, the conversion fails:
    1. resultType of tosa.slice is UnrankedTensorType
    2. tosa.slice.getsize().size() != resultType.getRank()

krzysz00 commented on mlir/test/Conversion/TosaToTensor/tosa-to-tensor-invalid.mlir:
```diff
@@ -0,0 +1,15 @@
+  // expected-error@+1 {{failed to legalize operation 'tosa.slice'}}
+  %0 = "tosa.slice"(%arg0) {start = array<i64: 2>, size = array<i64: 2, 3>} : (tensor<12xf32>)  -> (tensor<2xf32>)
+  return %0 : tensor<2xf32>
+}
```
Nit: missing newline

krzysz00 reviewed (approved):
LGTM, approved pending nits

LiqinWeng committed:
[MLIR][TOSA] Add tosa.slice operation conversion failure scenario

Fixes #68481, In the following scenario, the conversion fails:
1. resultType of tosa.slice is UnrankedTensorType
2. tosa.slice.getsize().size() < resultType.getRank()

LiqinWeng commented:
> LGTM, approved pending nits

Hello, my code was merged and the test case (tosa-to-tensor-invalid.mlir)  failed. There is no problem with my local test. I don’t know how to deal with it,(shold revert it ? ), Can you help me find out what caused it?

joker-eph commented:
Please revert until this is figured out.
